infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -946,6 +994,65 @@ private AddressMetaData writeRecord(SegmentHandle fh, long address,
         return new AddressMetaData(metadata.getChecksum(), metadata.getLength(), channelOffset);
     }
 
+    long getSegment(LogData entry) {
+        return entry.getGlobalAddress() / RECORDS_PER_LOG_FILE;
+    }
+
+    @Override
+    public void append(List<LogData> entries) {
+        // ignore empty list?
+
+        // check if the entries range cross a segment
+        LogData first = entries.get(0);
+        LogData last = entries.get(entries.size() - 1);
+        SegmentHandle firstSh = getSegmentHandleForAddress(first.getGlobalAddress());
+        SegmentHandle lastSh = getSegmentHandleForAddress(last.getGlobalAddress());
+
+        long segmentDiff = lastSh.getSegment() - firstSh.getSegment();
+
+        if (segmentDiff > 1) {
+            // Range overlaps more than two segments
+            throw new RuntimeException("Write range too large!");
+        }
+
+        List<LogData> segOneEntries = new ArrayList<>();
+        List<LogData> segTwoEntries = new ArrayList<>();
+        long base = first.getGlobalAddress();
+
+        for (int ind = 0; ind < entries.size(); ind++) {
+            LogData curr = entries.get(ind);
+
+            if (!curr.getGlobalAddress().equals(base + ind)) {
+                throw new RuntimeException("Entries not in sequential order!");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)